client/src/query_builder.rs

```diff
@@ -0,0 +1,58 @@
+use std::fmt::Debug;
+
+use iroha_data_model::{
+    predicate::PredicateBox,
+    query::{sorting::Sorting, Pagination, Query},
+    Value,
+};
+
+use crate::client::{Client, QueryOutput, QueryResult};
+
+pub struct QueryRequestBuilder<'a, R> {
+    client: &'a Client,
+    request: R,
+    pagination: Pagination,
+    filter: PredicateBox,
+    sorting: Sorting,
+}
+
+impl<'a, R> QueryRequestBuilder<'a, R>
+where
+    R: Query + Debug,
+    R::Output: QueryOutput,
+    <R::Output as TryFrom<Value>>::Error: Into<eyre::Error>,
+{
+    pub(crate) fn new(client: &'a Client, request: R) -> Self {
+        Self {
+            client,
+            request,
+            pagination: Pagination::default(),
+            sorting: Sorting::default(),
+            filter: PredicateBox::default(),
+        }
+    }
+
+    pub fn filter(mut self, filter: PredicateBox) -> Self {
+        self.filter = filter;
+        self
+    }
+
+    pub fn sort(mut self, sorting: Sorting) -> Self {
+        self.sorting = sorting;
+        self
+    }
+
+    pub fn paginate(mut self, pagination: Pagination) -> Self {
+        self.pagination = pagination;
+        self
+    }
+
+    pub fn request(self) -> QueryResult<<R::Output as QueryOutput>::Target> {
```

Comment:
What about using `execute` or something? In combination with the client methods it is `client.query(...).sort(...).filter(..).paginate(..).request()`, and `request` seems to duplicate `query` by sense.

Reply:
I don't think so. `execute` is less informative than `request` in my opinion.
Btw if we rename `query` to `build_query`, it will eliminate any confusion.

Reply:
`execute` is used in smart contracts. Either `request` or `execute` is fine for me, but I would like to see consistency

Reply:
ok let's rename it to `execute()`